### PR TITLE
feat: add support for network resource in a different subscription when using MSI

### PR DESCRIPTION
**Pull request by feiskyer**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

feat: add support for network resource in a different subscription when using MSI

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
feat: add support for network resource in a different subscription when using MSI
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/provider/azure_standard_test.go [hunk 1]
```diff
@@ -545,44 +545,54 @@ func testGetLoadBalancerSubResourceID(
 		resourceGroupName                   string
 		subResourceName                     string
 		useNetworkResourceInDifferentTenant bool
+		useNetworkResourceInDifferentSub    bool
 		expected                            string
 	}{
 		{
-			description:                         "resource id should contain NetworkResourceSubscriptionID when using network resources in different subscription",
+			description:                         "resource id should contain NetworkResourceSubscriptionID when using network resources in different tenant and ",
```
Comment (jwtty) [inline pkg/provider/azure_standard_test.go:552]: tenant and subscription?

**Comment by MartinForReal** [comment]
LGTM

File: pkg/auth/azure_auth.go [hunk 2]
```diff
@@ -248,12 +248,18 @@ func ParseAzureEnvironment(cloudName, resourceManagerEndpoint, identitySystem st
...
-// Return true when one of NetworkResourceTenantID and NetworkResourceSubscriptionID are specified
-// and equal to one defined in global configs
-func (config *AzureAuthConfig) UsesNetworkResourceInDifferentTenantOrSubscription() bool {
-	return (len(config.NetworkResourceTenantID) > 0 && !strings.EqualFold(config.NetworkResourceTenantID, config.TenantID)) ||
-		(len(config.NetworkResourceSubscriptionID) > 0 && !strings.EqualFold(config.NetworkResourceSubscriptionID, config.SubscriptionID))
+// UsesNetworkResourceInDifferentTenant determines whether the AzureAuthConfig indicates to use network resources in
+// different AAD Tenant than those for the cluster. Return true when NetworkResourceTenantID is specified  and not equal
+// to one defined in global configs
+func (config *AzureAuthConfig) UsesNetworkResourceInDifferentTenant() bool {
+	return (len(config.NetworkResourceTenantID) > 0 && !strings.EqualFold(config.NetworkResourceTenantID, config.TenantID))
```
Comment (lzhecheng) [inline pkg/auth/azure_auth.go:255]: nit: unnecessary outer parentheses

File: pkg/provider/azure_standard_test.go (hunk 1 above)
Comment (feiskyer) [reply pkg/provider/azure_standard_test.go]: good catch, fixed now.

File: pkg/auth/azure_auth.go (hunk 2 above)
Comment (feiskyer) [reply pkg/auth/azure_auth.go]: good catch, fixed now.

**Pull request merged**